cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/FieldLineageProcessor.java

```diff
@@ -124,13 +125,20 @@ public FieldLineageProcessor(PipelineSpec pipelineSpec) {
       if (invalidFieldOperations != null) {
         stageInvalids.put(stageName, invalidFieldOperations);
       }
+
+      if (!stageOperationsValidator.getRedundantOutputs().isEmpty()) {
+        stageRedundants.put(stageName, stageOperationsValidator.getRedundantOutputs());
+      }
+    }
+
+    if (!stageRedundants.isEmpty()) {
+      LOG.debug("The pipeline has redundant operations {} and they will be ignored", stageRedundants);
     }
 
     if (!stageInvalids.isEmpty()) {
       // Do not throw but just log the exception message for validation failure
       // Once most of the plugins are updated to write lineage exception can be thrown
       LOG.debug(new InvalidLineageException(stageInvalids).getMessage());
-      return Collections.emptySet();
```

Comment:
why is this change needed?

Reply:
This is to make sure we will process the rest field lineage. If we return an empty set here, the lineage will be gone. We want to make sure the valid field lineage is written.